### ISSUE TITLE
Concent does not validate reason in SubtaskResultsRejected

### ISSUE THREAD

**Issue by cameel** [golemfactory/concent#1140]
Concent should treat `SubtaskResultsRejected` message submitted by the requestor as invalid if the `reason` does not make sense. As stated in https://github.com/golemfactory/golem/issues/4217:
> There are only two valid reasons for the SRR:
> a) genuine `VerificationNegative` -> which will trigger Concent's UC4 later on
> b) a `ForcedResourcesFailure` in which case the message needs to contain a valid `ForceGetTaskResultFailed` signed by the Concent itself [ depends on https://github.com/golemfactory/golem-messages/issues/346 ]

Add validations for this case. 

Note that https://github.com/golemfactory/golem-messages/issues/346 referenced in b) will make it possible for the requestor to include `ForceGetTaskResultFailed` along with his `SubtaskResultsRejected`. This nested message (its presence, signature, content, etc.) must be validated along with the reason.

When you're done please also make sure that we validate reasons in other messages that contain a `reason` field. If we do not, report these cases as bugs.

**Comment by shadeofblue** [comment]
@cameel possibly those bugs could partially be addressed in golem-messages

**Comment by shadeofblue** [comment]
anyway, you should be unblocked here: https://github.com/golemfactory/golem-messages/releases/tag/v3.6.0

**Comment by cameel** [comment]
Whether the reason is present and has a valid value - yeah, that's something that's best done in GM. Only validating whether the reason is appropriate for the use case really needs to be in Concent.

**Comment by Jakub89** [comment]
@cameel @shadeofblue if only `VerificationNegative` and `ForcedResourcesFailure` are allowed for `SubtaskResultsRejected` in concent, what's the point for `ConcentResourcesFailure` and `ConcentVerificationNegative` REASONs Enum?

**Comment by cameel** [comment]
```
VerificationNegative        = 'Results verification negative'
ConcentResourcesFailure     = 'Concent could not retrieve resources to verify'
ConcentVerificationNegative = 'Concent results verification negative'
ForcedResourcesFailure      = 'Concent reported failure to retrieve the resources to verify'
ResourcesFailure            = 'Could not retrieve resources'
```

`ConcentResourcesFailure` and `ConcentVerificationNegative` are meant to be used in rejections issued by Concent. It's technically possible for the requestor to use them but if he does, his rejection does not make sense.

BTW, I just realized that this check could be made a validation in golem-messages. Currently we're just treating every reason as a separate thing and deciding which makes sense in this particular situation and which does not. But your question made me look at it from a different perspective: it never makes sense for requestor to use a reason meant for Concent and vice versa. So golem-messages could automatically reject some reasons by looking at who signed the message. What do you think @shadeofblue @dybi?

Another thing. I see that we're not using `ConcentVerificationNegative` anywhere in the code. I think it should be used in rejections from UC4. Please check.

> When you're done please also make sure that we validate reasons in other messages that contain a reason field. If we do not, report these cases as bugs.

@Jakub89 Have you done this part of the task? Please confirm.

**Comment by Jakub89** [comment]
@cameel I've done only simple verification for `SubtaskResultsRejected` #1143 . Let's do the rest of them in separate issue/PR after further discussion maybe?

**Comment by dybi** [comment]
@cameel @shadeofblue 
> So golem-messages could automatically reject some reasons by looking at who signed the message. What do you think @shadeofblue @dybi?

Seems reasonable. @shadeofblue is it possible to have it? And if so, who is going to implement it (CP or Golem Team)?
